drivers/windows/windows.go

```diff
@@ -726,7 +727,15 @@ func (d *driver) Join(nid, eid string, sboxKey string, jinfo driverapi.JoinInfo,
 		return err
 	}
 
-	// This is just a stub for now
+	endpoint.containerID = sboxKey
```

Comment:
As per the design, containerID is internal to libnetwork core while the drivers shouldnt assume the presence of container-id. SandboxKey is the generic concept that can be trusted without assuming any changes to the core.

Reply:
Yes, I understand that.
But libnetwork's design principle will not work well with Windows, because there is a service "HCS" that manages all the containers. So for any changes to the container, HCS has to be notified of that change. For this the ID of the container is required. This is the basic design principle of Windows Containers. 

To explain the internals more, Like CreateComputeSystem HCS call, for Hot Add/Remove of network endpoints, ModifyCompyteSystem HCS call has to be invoked.

Unlike Linux, on Windows, Containers and Networks are not yet decoupled completely to fit into LibNetwork design principle. Once that is done (am not sure when), we could drop the containerID. 

Till then windows needs containerID for hot add to work.